core/src/main/java/com/yahoo/oak/OakHashMap.java

```diff
@@ -0,0 +1,574 @@
+     * in ascending order of the corresponding keys.
+     */
+    private Iterator<V> valuesIterator() {
+        return internalOakMap.valuesTransformIterator(fromKey, fromInclusive, toKey, toInclusive, isDescending,
+                valueDeserializeTransformer);
+    }
+
+    /**
+     * Returns a {@link Iterator} of the mappings contained in this map in ascending key order.
+     */
+    private Iterator<Map.Entry<K, V>> entriesIterator() {
+        return internalOakMap.entriesTransformIterator(fromKey, fromInclusive, toKey, toInclusive, isDescending,
+                entryDeserializeTransformer);
+    }
+
+    /**
+     * Returns a {@link Iterator} of the keys contained in this map in ascending order.
+     */
+    private Iterator<K> keysIterator() {
+        return internalOakMap.keysTransformIterator(fromKey, fromInclusive, toKey, toInclusive, isDescending,
+                keyDeserializeTransformer);
+    }
+
+    private Iterator<OakUnscopedBuffer> keysBufferIterator() {
+        return internalOakMap.keysBufferViewIterator(fromKey, fromInclusive, toKey, toInclusive, isDescending);
+    }
+
+
+    private Iterator<OakUnscopedBuffer> valuesBufferIterator() {
+        return internalOakMap.valuesBufferViewIterator(fromKey, fromInclusive, toKey, toInclusive, isDescending);
+    }
+
+    private Iterator<Map.Entry<OakUnscopedBuffer, OakUnscopedBuffer>> entriesBufferIterator() {
+        return internalOakMap.entriesBufferViewIterator(fromKey, fromInclusive, toKey, toInclusive, isDescending);
+    }
+
+    private Iterator<OakUnscopedBuffer> keysStreamIterator() {
+        return internalOakMap.keysStreamIterator(fromKey, fromInclusive, toKey, toInclusive, isDescending);
+    }
+
+
+    private Iterator<OakUnscopedBuffer> valuesStreamIterator() {
+        return internalOakMap.valuesStreamIterator(fromKey, fromInclusive, toKey, toInclusive, isDescending);
+    }
+
+    private Iterator<Map.Entry<OakUnscopedBuffer, OakUnscopedBuffer>> entriesStreamIterator() {
+        return internalOakMap.entriesStreamIterator(fromKey, fromInclusive, toKey, toInclusive, isDescending);
+    }
+
+    /* ---------------- TODO: Move methods below to their proper place as they are implemented -------------- */
+
+
+    @Override
+    public void replaceAll(BiFunction<? super K, ? super V, ? extends V> function) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public V computeIfAbsent(K key, Function<? super K, ? extends V> mappingFunction) {
+        return null;
+    }
+
+    @Override
+    public V compute(K key, BiFunction<? super K, ? super V, ? extends V> remappingFunction) {
+        return null;
+    }
+
+    @Override
+    public V merge(K key, V value, BiFunction<? super V, ? super V, ? extends V> remappingFunction) {
+        return null;
+    }
+
+    /* ---------------- View Classes -------------- */
+
+    static class KeySet<K> extends AbstractSet<K>  {
```

Comment:
Can we reuse the code of `KeySet`, `EntrySet`, and the others from `OakMap`?

Reply:
not easily because they are inner classes  of the respective classes because  you wanted the iterators of the OakHashMap and OakMap functions to be private and be avilable only from the code of KeySet and such.